internal/core/data/init.go

```diff
@@ -166,7 +166,17 @@ func newDBClient(dbType string) (interfaces.DBClient, error) {
 			Host: Configuration.Databases["Primary"].Host,
 			Port: Configuration.Databases["Primary"].Port,
 		}
-		return redis.NewClient(dbConfig) // TODO: Verify this also connects to Redis
+		redisClient, err := redis.NewClient(dbConfig) // TODO: Verify this also connects to Redis
+		if err != nil {
+			return nil, err
+		}
+
+		// Background process for deleting device readings and events.
+		// This only needs to be running for core-data since this is the service responsible for handling the deletion
+		// of events
+		go redisClient.AsyncDeleteReadings(LoggingClient)
+		go redisClient.AsyncDeleteEvents(LoggingClient)
```

Comment:
Manage these inside of the redisClient itself.

```
// Client represents a Redis client
type Client struct {
    Pool *redis.Pool // A thread-safe pool of connections to Redis
    deleteReadingsChannel chan string
    deleteEventsChannel chan string
}

// Return a pointer to the Redis client
func NewClient(config db.Configuration) (*Client, error) {
    once.Do(func() {
    ...
        currClient = &Client{
            ...
            deleteReadingsChannel = make(chan string)
            deleteEventsChannel = make(chan string)
        }
        go currClient.AsyncDeleteReadings()
        go currClient.AsyncDeleteEvents()
    ...
    }
}
```
You should then close these channels in the RedisClient as part of its `CloseSession` function. Init.go calls `dbClient.CloseSession()` [here](https://github.com/edgexfoundry/edgex-go/blob/63c99229ab73133cb40a9f2e3e8640625d2de825/internal/core/data/init.go#L122) As it is now, the channels are never closed.

Reply:
Trevor and I discussed these changes via Zoom. We talked about the undesired effects these changes would have. If we implement these changes other services that use the Redis client will also spawn these go-routines which handle the deletion of readings and events, which will do nothing and just sit idle since other services are not responsible for that functionality. We talked about the possibility of refactoring the `NewClient` function to also accept an interface, lets call it `ChannelHandler` which will create the necessary channels and go-routines, there is some challenges and design decisions that will need to happen if we go down that road. I am going to take a stab at implementing what we discussed